go/obscuronode/node.go

```diff
@@ -286,35 +368,28 @@ func (a *Node) checkForSharedSecretRequests(block common.EncodedBlock) {
 	}
 }
 
-func NewAgg(
-	id common.NodeID,
-	cfg AggregatorCfg,
-	l1 common.L1Node,
-	l2Network L2Network,
-	collector StatsCollector,
-	genesis bool,
-) Node {
-	return Node{
-		// config
-		ID:        id,
-		cfg:       cfg,
-		mining:    true,
-		genesis:   genesis,
-		L1Node:    l1,
-		l2Network: l2Network,
-
-		stats: collector,
-
-		// lifecycle channels
-		exitNodeCh: make(chan bool),
-		interrupt:  new(int32),
+// processP2PRollups handles rollups gossiped by other obscuro nodes
+func (a *Node) processP2PRollups(rollup *obscuroCommon.Rollup) {
+	stored := a.Enclave.SubmitRollup(obscuroCommon.ExtRollup{
+		Header: rollup.Header,
+		Txs:    rollup.Transactions,
+	})
 
-		// incoming data
-		blockRPCCh:   make(chan blockAndParent),
-		forkRPCCh:    make(chan []common.EncodedBlock),
-		rollupsP2PCh: make(chan common.EncodedRollup),
+	height := uint(0)
```

Comment:
I don't think this logic is correct.
These are all non-confirmed rollups received via gossip. Only useful for the enclave to determine the winner. 
They don't need to be stored on the node

Reply:
I know exactly what you mean. I reckon there is a bug somewhere, but I can say for sure that if the gossiped rollups are not stored, the header traversing does not work. Let's chat about this :)